Core/include/Acts/Seeding/ISourceLinkGrid.hpp

```diff
@@ -0,0 +1,43 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/EventData/SourceLink.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Utilities/Axis.hpp"
+#include "Acts/Utilities/Grid.hpp"
+
+namespace Acts {
+
+/// @brief Interface class for binning source links
+/// into a lookup table -- a grid of source links
+///
+/// @tparam axis_t Type of the axis
+///
+template <typename axis_t>
+class ISourceLinkGrid {
```

Comment:
I don't understand this inheritance pattern. The base class is already templated on the axis type 

- btw it's a single axis, would you not want to be able to have different ones?
- is this ever held as the base where it cannot instead be held as the derived? You're already templating `PathSeeder` on the axis type, why not just template it on the whole "source link grid" instead and skip the inheritance?

Reply:
That's a remnant of me being obsessed with the interfaces. Changed to `Delegate`-based implementation to be consistent with the rest of the interface. Thanks for noticing that